packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -310,20 +320,32 @@ class _HeroState extends State<Hero> {
       'A Hero widget cannot be the descendant of another Hero widget.'
     );
 
-    if (_placeholderSize != null) {
-      if (widget.placeholderBuilder == null) {
-        return SizedBox(
-          width: _placeholderSize.width,
-          height: _placeholderSize.height,
-        );
-      } else {
-        return widget.placeholderBuilder(context, widget.child);
-      }
-    }
-    return KeyedSubtree(
+    final Widget keyedChild = KeyedSubtree(
       key: _key,
       child: widget.child,
     );
+
+    // If the hero is not in flight, build its child widget.
+    if (_placeholderSize == null) {
+      return keyedChild;
+    }
+
+    if (widget.placeholderBuilder != null) {
+      return widget.placeholderBuilder(context, widget.child);
+    }
+
+    return _shouldIncludeChild
+      ? Visibility(
+          visible: false,
+          maintainState: true,
+          maintainAnimation: true,
```

Comment:
ummm this is a bit unfortunate. @HansMuller it's not super clear to me why this is needed looking at the cascading logic in Visibility. Do you know?

Reply:
Not sure what you mean by "cascading logic". I agree that keeping the from Hero's animation running while it is hidden seems unnecessary.

Reply:
I just meant this series of logic https://github.com/flutter/flutter/blob/e2042bae6df774cc0be2b22d84f3510b93b24748/packages/flutter/lib/src/widgets/visibility.dart#L69

Reply:
Replacing the visibility widget with `SizedBox(Offstage(TickerMode))` seems to work